src/components/PhotoUpload/PhotoUploadCaptchaToolbar.js

```diff
@@ -0,0 +1,381 @@
+        flex-basis: 16.666666% !important;
+      }
+    }
+  }
+`;
+
+/* TODO: prepare stylable component 
+const contentBelowSearchField = styled.button`
+    overflow-x: hidden;
+    overflow-y: auto;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    
+    padding: 1em;
+    text-align: center;
+  
+    .captcha-content {
+    }
+
+    h3.captcha-help {
+    }
+    
+    small.captcha-explanation {
+
+    }
+  }
+`;
+*/
+
+export default class PhotoUploadCaptchaToolbar extends React.Component<Props, State> {
+  props: Props;
+
+  state: State = {
+    waitingForCaptcha: false,
+    captchaError: null,
+    captcha: null,
+  };
+
+  inputField: ?HTMLInputElement;
+  backLink: ?React.ElementRef<typeof CloseLink>;
+  goButton: ?React.ElementRef<'button'>;
+  checkCaptchaTimer: number | null;
+
+  componentDidMount() {
+    if (!this.props.hidden) {
+      this.startTimer();
+      this.focus(); // Focus input field on start
+    }
+
+    accessibilityCloudImageCache.getCaptcha().then(captcha => {
+      this.setState({waitingForCaptcha: false, captcha, captchaError: false});
+    }).catch(e => {
+      this.setState({captchaError: true});
+    });
+  }
+
+  componentWillUnmount() {
+    this.clearTimer();
+  }
+
+  componentWillReceiveProps(nextProps: Props) {
+    if (!nextProps.hidden) {
+      this.startTimer();
+    } else if (nextProps.hidden) {
+      this.clearTimer();
+    }
+  }
+
+  startTimer() {
+    if (!this.checkCaptchaTimer) {
+      this.refreshCaptcha();
+      this.checkCaptchaTimer = setInterval(this.refreshCaptcha, 5000);
+    }
+  }
+  
+  clearTimer() {
+    if (this.checkCaptchaTimer !== null) {
+      clearInterval(this.checkCaptchaTimer);
+      this.checkCaptchaTimer = null;
+    }
+  }
+
+  refreshCaptcha() {
+    if (!accessibilityCloudImageCache.hasValidCaptcha) {
+      this.setState({waitingForCaptcha: true});
+      accessibilityCloudImageCache.getCaptcha().then(captcha => {
+        this.setState({waitingForCaptcha: false, captcha, captchaError: false });
+      }).catch(e => {
+        this.setState({captchaError: true});
+      });
+    }
+  }
+
+  focus() {
+    this.inputField && this.inputField.focus();
+  }
+
+  renderInputField() {
+    const isInputDisabled = !this.canEnter();
+    return (
+      <input type="text"
+        ref={inputField => this.inputField = inputField}
+        onFocus={(event) => {
+          window.scrollTo(0, 0);  // Fix iOS mobile safari viewport out of screen bug
+        }}
+        disabled={isInputDisabled}
+        onChange={event => this.setState({ enteredCaptchaValue: event.target.value })}
+      />
+    );
+  }
+
+  renderBackLink() {
+    return <button
+      className='close-link'
+      // translator: Button caption in photo captcha dialog
+      aria-label={t`Back`}
+      onClick={() => {
+        if (this.props.onClose) this.props.onClose();
+      }}
+      ref={backLink => this.backLink = backLink}
+    >← </button>;
```

Comment:
One of the other PRs has a `<ChevronLeft />` if needed :)

Reply:
@nero-still something for you, maybe :)